.github/actions/make_init/action.yml

```diff
@@ -0,0 +1,77 @@
+name: "Initialize Virtual Env"
+runs:
+  using: composite
+  steps:
+    - name: Restore nvm, pre-commit cache, & node modules from cache
+      id: cache-nvm-node-modules
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.nvm
+          ~/.cache
+          frontend/node_modules
+        key: v13-nvm-node-modules-${{ hashFiles('**/yarn.lock.md5') }}-${{ hashFiles('**/.pre-commit-config.yaml') }}
+    - name: Install nvm, pre-commit cache and node_modules
+      if: steps.cache-nvm-node-modules.cache-hit != 'true'
+      run: |
+        if [ ! -d ~/.nvm ] ; then
+          cd ..
+          export NVM_DIR="$HOME/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+          mkdir $NVM_DIR
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+          echo "source $NVM_DIR/nvm.sh --install" >> "$HOME/.bash_profile"
+          cd -
+          source $HOME/.bash_profile
+          echo "lts/*" > .nvmrc
+          nvm install
+        fi
+        npm install --location=global yarn
+        if [ ! -d frontend/node_modules ] ; then
+          source "$HOME/.nvm/nvm.sh"
+          make react-init
+        fi
+      shell: bash -ileo pipefail {0}
+    - name: Install pyodbc dependencies
+      run: |
+        sudo apt-get install -y unixodbc-dev
+      shell: bash -ileo pipefail {0}
+    - name: Install graphviz dependencies
+      run: |
+        sudo apt update
+        sudo apt-get install -y libgvc6
```

Comment:
Splitting all of these dependency installations out into different stages seems like overkill to me. I feel like we might as well group all of them together and install them in a single `sudo apt install -y ...` command (along with the protobuf compiler that gets installed in a stage below).

This could look something like an "Install dependencies" stage that runs

```bash
sudo apt update
sudo apt install -y gnupg \
    unixodbc-dev \ # pyodbc dependencies
    libgvc6 \ # graphviz dependencies
echo "deb http://ppa.launchpad.net/maarten-fonville/protobuf/ubuntu trusty main" | sudo tee /etc/apt/sources.list.d/protobuf.list
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4DEA8909DC6A13A3
sudo apt update
sudo apt install -y protobuf-compiler
```

Depending on the conclusion of investigating what's going on with the `pre_make` action, we could also just drop that action entirely and install `graphviz` here too.

Reply:
Good call! Refactored to remove `pre_make` action and install all dependencies in one step 👍🏼